src/OrleansLoggerProvider/OrleansLoggingDecorator.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+
+namespace Orleans.Runtime
+{
+
+    /// <summary>
+    /// Config for message bulking feature
+    /// </summary>
+    public class MessageBulkingConfig
+    {
+        /// <summary>
+        /// Count limit for bulk message output.
+        /// If the same log code is written more than <c>BulkMessageLimit</c> times in the <c>BulkMessageInterval</c> time period, 
+        /// then only the first <c>BulkMessageLimit</c> individual messages will be written, plus a count of how bulk messages suppressed.
+        /// </summary>
+        public int BulkMessageLimit { get; set; } = DefaultBulkMessageLimit;
+
+        /// <summary>
+        /// Default bulk message limit. 
+        /// </summary>
+        public const int DefaultBulkMessageLimit = 5;
+
+        /// <summary>
+        /// Time limit for bulk message output.
+        /// If the same log code is written more than <c>BulkMessageLimit</c> times in the <c>BulkMessageInterval</c> time period, 
+        /// then only the first <c>BulkMessageLimit</c> individual messages will be written, plus a count of how bulk messages suppressed.
+        /// </summary>
+        public TimeSpan BulkMessageInterval { get; set; } = DefaultBulkMessageInterval;
+
+        /// <summary>
+        /// Default bulk message interval
+        /// </summary>
+        public static readonly TimeSpan DefaultBulkMessageInterval = TimeSpan.FromMinutes(1);
+    }
+
+    /// <summary>
+    /// OrleansLoggingDecorator class. User can plug in their own ILogger implementation into this decorator class to add message bulking feature on top of their logger. 
+    /// Message bulking feature will just log eventId count if the same eventId has appear more than BulkMessageLimit in a certain BulkMessageInterval.
+    /// </summary>
+    public class OrleansLoggingDecorator : ILogger
```

Comment:
I'd rename to `MessageBulkingDecoratorLogger` or even (`EventBulkingDecoratorLogger`), as there is nothing inherent to Orleans nor the legacy logger. In theory this could be externalized as a separate package that non-Orleans users could use (not saying we should do the latter now though, just mentioning how decoupled this is).

Reply:
will do